### PR TITLE
Clean Up `theia.d.ts`

### PR DESCRIPTION
**Pull request by colin-grant-work**
Depends on #11564.

#### What it does

There are a number of syntactic (misplaced `readonly` modifiers) and semantic (use of undeclared variables) problems in our `theia.d.ts` file. This cleans them up.

We can detect these problems by setting `skipLibCheck: false` in `packages/plugin/tsconfig.json`, but that makes it impossible to compile the application because we have a lot of conflicting declarations in our node modules.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/main/browser/authentication-main.ts [hunk 1]
```diff
@@ -273,7 +254,7 @@ interface AccountUsage {
 
 export class AuthenticationProviderImpl implements AuthenticationProvider {
     private accounts = new Map<string, string[]>(); // Map account name to session ids
-    private sessions = new Map<string, string>(); // Map account id to name
+    private sessions = new Map<string, string>();   // Map account id to name
```
Comment (paul-marechal) [inline packages/plugin-ext/src/main/browser/authentication-main.ts:257]: nitpick: unnecessary change.

File: packages/plugin/src/theia-proposed.d.ts [hunk 2]
```diff
@@ -705,3 +610,21 @@ export module '@theia/plugin' {
...
     // #endregion
 }
+
+/**
+ * Thenable is a common denominator between ES6 promises, Q, jquery.Deferred, WinJS.Promise,
+ * and others. This API makes no assumption about what promise library is being used which
+ * enables reusing existing code without migrating to a specific promise implementation. Still,
+ * we recommend the use of native promises which are available in this editor.
+ */
+interface Thenable<T> {
```
Comment (paul-marechal) [inline packages/plugin/src/theia-proposed.d.ts:620]: Isn't this type already defined in `@theia/plugin`? Is it necessary to re-declare it here?

**Review by paul-marechal — COMMENTED** [review]
Looks mostly good, thanks for the cleanup!

File: packages/plugin/src/theia-proposed.d.ts (hunk 2 above)
Comment (colin-grant-work) [reply packages/plugin/src/theia-proposed.d.ts:620]: I'm open to suggestions. It could be declared only in `theia.proposed.d.ts` and then picked up automatically in `theia.d.ts` but it's good for it to be in `theia.d.ts` for the use of the comparator.

File: packages/plugin/src/theia.d.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ import './theia-proposed';
 /* eslint-disable @typescript-eslint/no-explicit-any */
 /* eslint-disable max-len */
 
-export module '@theia/plugin' {
+declare module '@theia/plugin' {
```
Comment (vince-fugnitto) [inline packages/plugin/src/theia.d.ts:28]: This update unfortunately breaks our typedocs generation for `@theia/plugin` (fixed by https://github.com/eclipse-theia/theia/pull/10274). We may need to revisit the typedocs fix since it now produces a report with these changes like:

typedocs

**Review by vince-fugnitto — COMMENTED** [review]
The changes overall look good to me 👍 The only thing I noticed was how our typedocs generation would fail to produce any meaningful content now for the `@theia/plugin` extension.

**Review by vince-fugnitto — APPROVED** [review]
LGTM 👍 It'll definitely help with all the false positives we used to see and would now be correctly identified with the comparator updates.
